pkg/v1/tkg/test/tkgctl/shared/e2e_common_cc.go

```diff
@@ -0,0 +1,280 @@
+		namespace = constants.DefaultNamespace
+		input = inputGetter()
+		if input.Namespace != "" {
+			namespace = input.Namespace
+		}
+
+		logsDir = filepath.Join(input.ArtifactsFolder, "logs")
+
+		rand.Seed(time.Now().UnixNano())
+		clusterName = input.E2EConfig.ClusterPrefix + "wc-" + util.RandomString(4) // nolint:gomnd
+
+		tkgCtlClient, err = tkgctl.New(tkgctl.Options{
+			ConfigDir: input.E2EConfig.TkgConfigDir,
+			LogOptions: tkgctl.LoggingOptions{
+				File:      filepath.Join(logsDir, clusterName+".log"),
+				Verbosity: input.E2EConfig.TkgCliLogLevel,
+			},
+		})
+		Expect(err).To(BeNil())
+
+		mngContextName = input.E2EConfig.ManagementClusterName + "-admin@" + input.E2EConfig.ManagementClusterName
+		mngProxy = framework.NewClusterProxy(input.E2EConfig.ManagementClusterName, "", mngContextName)
+		mngKubeConfigFileName = input.E2EConfig.ManagementClusterName + ".kubeconfig"
+		mngKubeConfigFile = filepath.Join(os.TempDir(), mngKubeConfigFileName)
+
+		err = tkgCtlClient.GetCredentials(tkgctl.GetWorkloadClusterCredentialsOptions{
+			ClusterName: input.E2EConfig.ManagementClusterName,
+			Namespace:   "tkg-system",
+			ExportFile:  mngKubeConfigFile,
+		})
+		Expect(err).To(BeNil())
+
+		options = framework.CreateClusterOptions{
+			ClusterName:  clusterName,
+			Namespace:    namespace,
+			Plan:         "dev",
+			CniType:      input.Cni,
+			OtherConfigs: input.OtherConfigs,
+		}
+
+		if input.Plan != "" {
+			options.Plan = input.Plan
+		}
+
+		if input.E2EConfig.InfrastructureName == "vsphere" {
+			if input.Cni == "antrea" {
+				if clusterIP, ok := os.LookupEnv("CLUSTER_ENDPOINT_ANTREA"); ok {
+					options.VsphereControlPlaneEndpoint = clusterIP
+				}
+			}
+			if input.Cni == "calico" {
+				if clusterIP, ok := os.LookupEnv("CLUSTER_ENDPOINT_CALICO"); ok {
+					options.VsphereControlPlaneEndpoint = clusterIP
+				}
+			}
+		}
+		clusterConfigFile, err = framework.GetTempClusterConfigFile(input.E2EConfig.TkgClusterConfigPath, &options)
+		Expect(err).To(BeNil())
+
+		pacificCluster, err := tkgCtlClient.IsPacificRegionalCluster()
+		Expect(err).NotTo(HaveOccurred())
+		if pacificCluster {
+			infrastructureName = "TKGS"
+		} else {
+			infrastructureName = input.E2EConfig.InfrastructureName
+		}
+
+		By(fmt.Sprintf("Get k8s client for management cluster %q", input.E2EConfig.ManagementClusterName))
+		mngClient, mngDynamicClient, mngAggregatedAPIResourcesClient, mngDiscoveryClient, err = GetClients(ctx, mngKubeConfigFile)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		By(fmt.Sprintf("Deleting workload cluster %q", clusterName))
+		err = tkgCtlClient.DeleteCluster(tkgctl.DeleteClustersOptions{
+			ClusterName: clusterName,
+			Namespace:   namespace,
+			SkipPrompt:  true,
+		})
+		Expect(err).To(BeNil())
+
+		// verify addon resources are deleted successfully in clusterclass mode
+		if isClusterClass, ok := input.OtherConfigs["clusterclass"]; ok && isClusterClass == "true" {
```

Comment:
nit: shall we get rid of `input.OtherConfigs["clusterclass"]` here since we have individual test suite for CC? Also might be able to remove it from the test entries in `aws_cc` folder.

Reply:
Fair enough, I'll see what I can drop here